core/src/main/java/nl/nn/adapterframework/configuration/ConfigurationUtils.java

```diff
@@ -323,23 +326,19 @@ public String getVersion() {
 		}
 	}
 
-	public static String processMultiConfigZipFile(IbisContext ibisContext, String datasource, boolean activate_config, boolean automatic_reload, InputStream file, String ruser) throws IOException, ConfigurationException {
-		String result = "";
+	public static Map<String, Object> processMultiConfigZipFile(IbisContext ibisContext, String datasource, boolean activate_config, boolean automatic_reload, InputStream file, String ruser) throws IOException, ConfigurationException {
+		Map<String, Object> result = new HashMap<String, Object>();
```

Comment:
I think you can use a Map<String,String>
and instead of a HashMap you can use a LinkedHashMap, that uses the original order of insertion when iterated over.